lib/wallet/wallet.js

```diff
@@ -2569,6 +2570,157 @@ class Wallet extends EventEmitter {
     return mtx;
   }
 
+  async processNameForFinish(name, resource, acct) {
+    if (!rules.verifyName(name))
+      throw new Error(`Invalid name: "${name}".`);
+
+    const rawName = Buffer.from(name, 'ascii');
+    const nameHash = rules.hashName(rawName);
+    const ns = await this.getNameState(nameHash);
+    const height = this.wdb.height + 1;
+    const network = this.network;
+
+    if (!ns)
+      throw new Error(`Auction not found: "${name}".`);
+
+    if (ns.isExpired(height, network))
+      throw new Error(`Name has expired: "${name}"!`);
+
+    const state = ns.state(height, network);
+
+    if (state < states.CLOSED)
+      throw new Error(`Auction is not yet closed: "${name}".`);
+
+    const reveals = await this.txdb.getReveals(nameHash);
+    const outpoints = [];// { outpoint, output }
+
+    for (const {prevout, own} of reveals) {
+      const {hash, index} = prevout;
+
+      if (!own)
+        continue;
+
+      const coin = await this.getCoin(hash, index);
+
+      if (!coin)
+        continue;
+
+      if (acct != null && !await this.txdb.hasCoinByAccount(acct, hash, index))
+        continue;
+
+      // Is local?
+      if (coin.height < ns.height)
+        continue;
+
+      const output = new Output();
+      output.address = coin.address;
```

Comment:
This is a consensus requirement for REGISTER (see `lib/covenants/rules.js:1282`) but not for REDEEMs. @rozaydin can you confirm that the previous behavior for normal REDEEMs was to send the REDEEM output to the same address and not to a new address? I can imagine wanting to send the coins to a different address since REDEEMs are spendable. In any case not a blocking change